core/templates/services/math-interactions.service.ts

```diff
@@ -523,7 +523,30 @@ export class MathInteractionsService {
 
   // The input terms to this function should be the terms split by '+'/'-'
   // from an expression.
-  termsMatch(term1: string, term2: string): boolean {
+  doTermsMatch(term1: string, term2: string): boolean {
+    let doPartsMatch = function(part1: string, part2: string): boolean {
+      // Splitting a term by '*' or '/' would give a list of parts.
+      // Nerdamer simplifies constant terms automatically so to avoid that,
+      // we'll replace all constants with a variable.
+      // Multiple instances of the same constant will be replaced by the same
+      // variable.
+
+      // Replacing decimals with variables.
+      // Eg: 3.4 + x => const3point4 + x
+      // We need to do this differently since const3.4 would be
+      // an invalid variable.
+      part1 = part1.replace(/(\d+)\.(\d+)/g, 'const$1point$2');
+      part2 = part2.replace(/(\d+)\.(\d+)/g, 'const$1point$2');
+
+      // Replacing integers with variables.
+      // Eg: 2 + x * 4 + 2 => const2 + x * const4 + const2
```

Comment:
One check -- Nerdamer doesn't automatically simplify const2 + const2 to 2 * const2, right?

Reply:
It does, but they still won't match since the number of terms is different.